src/core/operations/YaraRules.mjs

```diff
@@ -0,0 +1,82 @@
+/**
+ * @author Matt C [matt@artemisbot.uk]
+ * @copyright Crown Copyright 2019
+ * @license Apache-2.0
+ */
+
+import Operation from "../Operation";
+import OperationError from "../errors/OperationError";
+import Yara from "libyara-wasm";
+
+/**
+ * Yara Rules operation
+ */
+class YaraRules extends Operation {
+
+    /**
+     * YaraRules constructor
+     */
+    constructor() {
+        super();
+
+        this.name = "Yara Rules";
+        this.module = "Yara";
+        this.description = "Yara support";
+        this.infoURL = "https://en.wikipedia.org/wiki/YARA";
+        this.inputType = "ArrayBuffer";
+        this.outputType = "string";
+        this.args = [{
+            name: "Rules",
+            type: "code",
+            value: ""
+        }];
+    }
+
+    /**
+     * @param {string} input
+     * @param {Object[]} args
+     * @returns {string}
+     */
+    run(input, args) {
+        return new Promise((resolve, reject) => {
+            Yara().then(yara => {
+                let matchString = "";
+                const inpArr = new Uint8Array(input); // I know this is garbage but it's like 1.5 times faster
+                const inpVec = new yara.vectorChar();
+                for (let i = 0; i < inpArr.length; i++) {
+                    inpVec.push_back(inpArr[i]);
+                }
+                const resp = yara.run(inpVec, args[0]);
+                if (resp.compileErrors.size() > 0) {
+                    for (let i = 0; i < resp.compileErrors.size(); i++) {
```

Comment:
I think you can use a `for ... of ...` loop here which might be slightly neater.

Reply:
Fair enough! I imagine `for ... in ...` would still work, but it's not a big deal.